omod-1.8/src/main/java/org/openmrs/module/webservices/rest/web/v1_0/resource/openmrs1_8/PatientIdentifierResource1_8.java

```diff
@@ -214,17 +214,29 @@ public PatientIdentifier getByUniqueId(String uniqueId) {
 	 */
 	@Override
 	public PatientIdentifier save(PatientIdentifier delegate) {
-		// make sure it has already been added to the patient
 		boolean needToAdd = true;
 		for (PatientIdentifier pi : delegate.getPatient().getActiveIdentifiers()) {
 			if (pi.equals(delegate)) {
 				needToAdd = false;
 				break;
 			}
 		}
-		if (needToAdd)
+		
+		if (needToAdd) {
 			delegate.getPatient().addIdentifier(delegate);
-		return service().savePatientIdentifier(delegate);
+		}
+		service().savePatientIdentifier(delegate);
+		//service().savePatient(delegate.getPatient());
+		
+		if (delegate.isPreferred()) {
+			for (PatientIdentifier pi : delegate.getPatient().getActiveIdentifiers()) {
+				if (!pi.equals(delegate)) {
+					pi.setPreferred(false);
+				}
```

Comment:
The changes you have in PatientIdentifierResource1_8.java look more than the ones for PersonAddressResource1_8.java in this commit https://github.com/openmrs/openmrs-module-webservices.rest/commit/d23b7dd48dc12c84d2b20e9749547d2e85e2fd63
Can you please do exactly the changes like Darius committed?

Reply:
ok. Thank you. !Great tip